### ISSUE TITLE
[Game Crash] "engine error: portalnum > numareaportals"

### ISSUE THREAD

**Issue by ReikaKalseki** [BEEmod/BEE2.4#537]
I made a small tweak to one of my maps - changing a button type from "weighted" to cube - and suddenly the game crashes the moment compilation finishes (and the map is loaded). The map is BTS-styled, but it worked fine until literally right before this change. I doubt the change actually means anything in and of itself, but it appears to have triggered a deeper engine issue.

![](http://i.imgur.com/2o8nrds.png)

EDIT:
I am now getting that error trying to load a different map - i.e. it crashes *loading the editor*. This map is NOT BTS styled - though it does have the same skybox (Factory 3D) - and similarly worked until moments earlier. This time it was not even changed.

**Comment by TeamSpen210** [comment]
Make sure you set it to use the 1st entry and exit corridors. I haven't made most of them yet.

**Comment by ReikaKalseki** [comment]
I left it on random. And it has worked for 90% of the levels thus far.

**Comment by LautaroL20** [comment]
This is a duplicate of: https://github.com/BEEmod/BEE2-items/issues/758

**Comment by ReikaKalseki** [comment]
Almost. Yours appears to be a result of the entry hallways available in BTS. But I am getting this in clean style maps (though with the factory skybox).

**Comment by LautaroL20** [comment]
Okay, then it looks like the factory skybox may cause this weird error with other styles too. Your issue has been moved to the item´s page: https://github.com/BEEmod/BEE2-items/issues/1635

**Comment by ReikaKalseki** [comment]
It has started bleeding into other styles and maps, including 1950s with the sphere interior skybox and clean with the corresponding skybox.

**Comment by ReikaKalseki** [comment]
Some testing implies it is chamber-specific; have my maps become corrupted?

**Comment by TeamSpen210** [comment]
It might be to do with the specific layout of the map, yes. (But it shouldn't be possible to layout maps to cause it.) Enable 'developer 2' in the console, run the compile through the first two stages, then cancel and check for errors in the new log.

**Comment by ReikaKalseki** [comment]
> Enable 'developer 2' in the console, run the compile through the first two stages, then cancel and check for errors in the new log.

How do I do this?


Also, if it helps, I installed Hammer and opened two of the three big trouble levels with it (2 images one track, the rest the other, most recent errorer). I do seem some oddities like objects looking like holes and purple-black missingtex.

http://imgur.com/a/lJZku

**Comment by TeamSpen210** [comment]
Go in game options -> keyboard -> advanced and you can enable it. Then press ~ to toggle. That map's a false alarm it's what the editor produces before the BEE2 compiler messes with it. Open the one in the `styled/` folder. The compiler changes the instance paths to ones that work. The goop is missing-texture because it's a special texture name referring to a real-time render texture. The external walls are cleaned up by the real compiler, so it was easier for Valve to code the editor to do that.

You could also try Map -> Load Pointfile in Hammer, and see if it gives you a default. That will give you a red line in the views which should show the problem area. (It's basically either a part of the map 'outside'.)

**Comment by ReikaKalseki** [comment]
I cannot get *any* of the erroring maps to load in the PETI editor anymore - not even ones which only errored on compilation before.

As for the map in /styled - I opened it and most of the oddities are gone, though I do notice a small nodraw area full of cubes in the void and doorways full of "hint" walls (which googling the engine error pulled up mentions of).

I tried opening a pointfile, but there are no files of the appropriate extension anywhere in the folder structure for it to open.
